internal/pkg/daemon/enricher/filter.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package enricher
+
+import (
+	"encoding/json"
+	"reflect"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/go-logr/logr"
+
+	"sigs.k8s.io/security-profiles-operator/internal/pkg/daemon/enricher/types"
+)
+
+func GetEnricherFilters(enricherFiltersJsonStr string, logger logr.Logger) ([]types.EnricherFilterOptions, error) {
+	var enricherFilters []types.EnricherFilterOptions
+	if err := json.Unmarshal([]byte(enricherFiltersJsonStr), &enricherFilters); err != nil {
+		logger.Error(err, "failed to unmarshal enricher filters",
+			"enricherFiltersJsonStr", enricherFiltersJsonStr,
+			"err", err)
```

Comment:
I don't think we have to add the error twice:
```suggestion
			"enricherFiltersJsonStr", enricherFiltersJsonStr)
```